controllers/storageautoscaler/scraper.go

```diff
@@ -107,16 +113,23 @@ func filterObjectsForScaling(storageAutoScalerList *ocsv1.StorageAutoScalerList,
 			err := fmt.Errorf("osd usage is nil for device class, device class: %s", storageAutoScaler.Spec.DeviceClass)
 			log.Error(err, "device class not found in sync map")
 		} else {
-			deviceClassUsage := usage.(float64)
-			if (deviceClassUsage * 100) >= float64(deviceClassThreshold) {
-				objectList = append(objectList, types.NamespacedName{Namespace: storageAutoScaler.Namespace, Name: storageAutoScaler.Name})
+			deviceClassUsage := usage.(OsdUsage)
+			if (deviceClassUsage.HighestUsage * 100) >= float64(deviceClassThreshold) {
+				if totalUsageGreaterThanThresholdMinusTenPercent(deviceClassUsage, deviceClassThreshold) {
+					objectList = append(objectList, types.NamespacedName{Namespace: storageAutoScaler.Namespace, Name: storageAutoScaler.Name})
+				}
+				log.Info("highest osd usage is greater than the threshold, but total cluster usage is less than the (threshold-10)%, skipping scaling", "device class", deviceClass, "highest osd usage", deviceClassUsage.HighestUsage*100, "total osd usage", deviceClassUsage.TotalUsage*100)
 			}
 		}
 	}
 
 	return objectList
 }
 
+func totalUsageGreaterThanThresholdMinusTenPercent(deviceClassUsage OsdUsage, deviceClassThreshold int) bool {
```

Comment:
nit: don't bake the 10% threshold into the method name
```suggestion
func totalUsageGreaterThanAdjustedThreshold(deviceClassUsage OsdUsage, deviceClassThreshold int) bool {
```